services/comprehensive_blog_generator.py

```
@@ -0,0 +1,730 @@
+        except TokenLimitExceededError as e:
+            logger.error(f"Token limit exceeded for {date}: {e}")
+            # Try to reduce prompt size and retry
+            return self._handle_token_limit_exceeded(date, twitch_clips, github_events, e)
+        except AIClientError as e:
+            logger.error(f"AI generation failed for {date}: {e}")
+            raise
+        except Exception as e:
+            logger.error(f"Unexpected error in comprehensive blog generation: {e}")
+            raise AIClientError(f"Comprehensive blog generation failed: {e}")
+    
+    def _handle_token_limit_exceeded(self, date: str, twitch_clips: List[Dict[str, Any]], github_events: List[Dict[str, Any]], error: TokenLimitExceededError) -> Dict[str, Any]:
```

Ruff (0.12.2)

Unused method argument: `error`

(ARG002)